framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -415,7 +401,36 @@ public void standardizeReturnType(MethodTree methodDeclTree, AnnotatedTypeMirror
         JavaExpressionContext context =
                 JavaExpressionContext.buildContextForMethodDeclaration(
                         methodDeclTree, pathToMethodDecl, factory.getChecker());
-        standardizeDoNotUseLocalScope(context, pathToMethodDecl, atm);
+        viewpointAdaptToContext(context, atm);
+    }
+
+    /**
+     * Viewpoint-adapt a type qualifier obtained from a contract to {@code jeContext}.
+     *
+     * @param annoFromContract a qualifier that was written on a method declaration
+     * @param jeContext the context to use for standardization
+     * @param errorTree if non-null, where to report any errors that occur will parsing the
+     *     dependent type annotation; if null, report no errors
+     * @return the viewpoint-adapted annotation, or the argument if it is not a dependant type
+     *     annotation
+     */
+    public AnnotationMirror viewpointAdaptQualifierFromContract(
+            AnnotationMirror annoFromContract,
+            JavaExpressionContext jeContext,
+            @Nullable Tree errorTree) {
+        if (!hasDependentAnnotations()) {
+            return annoFromContract;
+        }
+
+        AnnotationMirror standardized =
+                standardizeAnnotationIfDependentType(jeContext, null, annoFromContract, false);
```

Comment:
The documentation of `standardizeAnnotationIfDependentType` doesn't say anything about viewpoint-adaptation, and the method of this body doesn't seem to have any other place that viewpoint-adaptation could be happening.

Reply:
I corrected the documentation of `standardizeAnnotationIfDependentType`; it does viewpoint-adapt.   Almost all of occurrences of "standardized" need to be changed to "viewpoint-adpat" in this class.  I'm going to defer this until JavaExpressions are viewpoint-adapted rather than strings.